src/nodes.coffee

```diff
@@ -3498,6 +3506,35 @@ exports.Op = class Op extends Base
   toString: (idt) ->
     super idt, @constructor.name + ' ' + @operator
 
+  ast: ->
+    @checkUpdateAssignability()
+    super()
+
+  astType: ->
+    switch @operator
+      when 'new'           then 'NewExpression'
+      when '||', '&&', '?' then 'LogicalExpression'
+      when '++', '--'      then 'UpdateExpression'
+      else
+        if @isUnary()      then 'UnaryExpression'
+        else                    'BinaryExpression'
+
+  astProperties: ->
+    firstAst = @first.ast()
+    secondAst = @second?.ast()
+    switch
+      when @operator is 'new'
+        callee: firstAst
```

Comment:
Again, this "smells" like `new` should be a `Call`, since here we're mimicking the AST of a call (and there'll end up being duplication of specifying `type: 'NewExpression'` since `new` with args is in fact a `Call`)

Reply:
I’ve been putting `return` before indented objects, to make clearer what’s happening:
```coffee
    switch
      when @operator is 'new'
        return
          callee: firstAst
          arguments: []
```

I know it’s not strictly necessary, but we don’t have any “loose” objects getting implicitly returned anywhere else in the codebase as far as I’m aware.